ci/functional/test_main_prep_node.sh

```diff
@@ -2,33 +2,220 @@
+if [ $opa_count -gt 0 ]; then
+    ((ib_count=opa_count)) || true
+fi
+
+while IFS= read -r line; do
+    mlnx_type="${line##*-}"
+    mlnx_type="${mlnx_type%*]}"
+    if [ "$mlnx_type" -ge 6 ]; then
+        ((hdr_count++)) || true
+    fi
+done < <(lspci -mm | grep "ConnectX")
+if [ $hdr_count -gt 0 ]; then
+    ((ib_count=hdr_count)) || true
+fi
+
+# Can not have Omni-Path and Mellanox HDR on the same system.
+if [ $hdr_count -gt 0 ] && [ $opa_count -gt 0 ]; then
+    ib_message="Invalid hardware configuration.  Found:
+$hdr_count Mellanox HDR ConnectX adapters,
+and
+$opa_count Omni-Path adapters.
+The Onmi-Path adapters will not be used."
+    mail_message+="${nl}${ib_message}${nl}"
+    echo "$ib_message"
+fi
+set -x
+
+function do_wait_for_ib {
+    ib_timeout=300 # 5 minutes
+    retry_wait=10 # seconds
+    timeout=$((SECONDS + ib_timeout))
+    while [ "$SECONDS" -lt "$timeout" ]; do
+      ip link set up "$1" || true
+      sleep 2
+      if ip addr show "$1" | grep "inet "; then
+        return 0
+      fi
+      sleep ${retry_wait}
+    done
+    return 1
+}
+
+# First check for infinband devices
+for i in $(seq 0 $((ib_count-1))); do
     iface="ib$i"
-    if [ -e /sys/class/net/"$iface" ]; then
-        if ! ifconfig "$iface" | grep "inet "; then
-          {
-            echo "Found interface $iface down after reboot on $HOSTNAME"
-            systemctl status || true
-            systemctl --failed || true
-            journalctl -n 500 || true
-            ifconfig "$iface" || true
-            cat /sys/class/net/"$iface"/mode || true
-            ifup "$iface" || true
-            ifconfig -a || true
-            ls -l /etc/sysconfig/network-scripts/ifcfg-* || true
-            cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
-          } 2>&1 | mail -s "Interface $iface found down after reboot" \
-                   -r "$HOSTNAME"@intel.com "$OPERATIONS_EMAIL"
+    if do_wait_for_ib "$iface"; then
+        set +x
+        if ! ip addr show "$iface" | grep "inet "; then
+            ib_message="$({
+                echo "Found interface $iface down after reboot on $HOSTNAME."
+                ip addr show "$iface" || true
+                cat /sys/class/net/"$iface"/mode || true
+                ip link set up "$iface" || true
+                cat /etc/sysconfig/network-scripts/ifcfg-"$iface" || true
+                } 2>&1)"
+            mail_message+="${nl}${ib_message}${nl}"
+            echo "$ib_message"
+            result=1
         fi
-        if ! ifconfig "$iface" | grep "inet "; then
-            echo "Failed to bring up interface $iface on $HOSTNAME. " \
-            "Please file a CORCI ticket."
-            exit 1
+        else
+            echo "OK: Interface $iface is up."
         fi
+        if [ -e "/sys/class/net/$iface/device/numa_node" ]; then
+            set -x
+            cat "/sys/class/net/$iface/device/numa_node"
+        fi
+        set -x
+    else
```

Comment:
(lint) Expected 'done' matching previously mentioned 'do'. [SC1062]
(lint) Unexpected keyword/token. Fix any mentioned problems and try again. [SC1072]